### DISCUSSION TITLE
An error occurs in Post processing. where is the cause?

### DISCUSSION THREAD

**Discussion by furuhashitcb** [codemagic-ci-cd/codemagic-docs#1042] [Q&A] [answered]
The app was created with Flutter.
So, CodeMagic build environment is Flutter workflow editor.
I tried uploaded version is 1.0.0(pubspec.yaml).

The error message below is displayed.
Please tell me how to fix it.
-------------------------
App Store distribution
-------------------------
Find application entry from App Store Connect for uploaded binary
Found 1 App matching specified filters: bundleId=com.XXXX.XXXXX.
-- App --
Id: XXXXXXXXXXX
Type: apps
Bundle id: com.XXXX.XXXXX
Name: XXXXXXXXXXXXX
Primary locale: en-US
Sku: XXXXXXXXXX
Available in new territories: False
Is or ever was made for kids: False

Find freshly uploaded build
Did not find build matching uploaded version yet, it might be still processing. Waiting to try again...
Did not find build matching uploaded version yet, it might be still processing. Waiting to try again...
The application binary was successfully uploaded to App Store Connect for processing but the corresponding build is not available in App Store Connect. This usually happens when the uploaded application has one or more issues. Please check your email for direct feedback from Apple.

**Comment by himesh-cm** [comment, marked as answer]
@furuhashitcb As suggested in the logs, did you check your email from apple? What does it say?

**Reply by furuhashitcb** [reply]
Thank you for your reply. I didn't see an email from APPLE. I received an email. The contents are below.
--------------------------------------------------------------------
Dear Developer,

We identified one or more issues with a recent delivery for your app, "XXXXXXXXXXXXX" 1.0.0 (1.0.0). Please correct the following issues, then upload again.

ITMS-90683: Missing Purpose String in Info.plist - Your app‘s code references one or more APIs that access sensitive user data. The app‘s Info.plist file should contain a NSPhotoLibraryUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. If you're using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. For details, visit: https://developer.apple.com/documentation/uikit/protecting_the_user_s_privacy/requesting_access_to_protected_resources

Though you are not required to fix the following issues, we wanted to make you aware of them:

ITMS-90078: Missing Push Notification Entitlement - Your app appears to register with the Apple Push Notification service, but the app signature's entitlements do not include the 'aps-environment' entitlement. If your app uses the Apple Push Notification service, make sure your App ID is enabled for Push Notification in the Provisioning Portal, and resubmit after signing your app with a Distribution provisioning profile that includes the 'aps-environment' entitlement. Xcode does not automatically copy the aps-environment entitlement from provisioning profiles at build time. This behavior is intentional. To use this entitlement, either enable Push Notifications in the project editor's Capabilities pane, or manually add the entitlement to your entitlements file. For more information, see https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#//apple_ref/doc/uid/TP40008194-CH6-SW1.

Best regards,

The App Store Team

**Reply by furuhashitcb** [reply]
Thanks to you, I found that there was a problem with my app settings. No problem on CodeMagic side.

**Reply by hzfplay** [reply]
@furuhashitcb How was the problem solved there? I also encountered this problem

**Reply by btanev** [reply]
I got the same error message recently - build is successful but App Store distribution fails with this message.
I've rerun the release build pipeline twice but didn't received any email from Apple. Have you encountered this scenario - is an issue in codemagic or apple?

The same pipeline was working as expected before the update to Xcode 26.

**EDIT**:
To anyone that finds this discussion with the same problem, here is a workaround:
- Download the release IPA from codemagic
- Deliver it to AppStore with the "Transporter" app

**Reply by dtrdic** [reply]
@btanev that’s not a good workaround. Apple rejected the app for a specific reason, which is mentioned in the failing log step. Do you mind creating a new discussion with a clear issue description so we can track it properly? Someone from our team will then take a look. Please also share the build ID related to the issue.

**Reply by btanev** [reply]
Hey there @dtrdic,
- I didn't get any messages from Apple on email or App Store Connect.
- Once I've uploaded the codemagic build with Transporter app it showed on AppStore Connect and I've submitted it for review - already published on the AppStore.

I assumed it was some kind of AppStore connect of codemagic issue - although that status pages showed no issues.

**Reply by dtrdic** [reply]
@btanev the post-processing error will tell you why the app was rejected. Can't tell the reason if you don't share any details
